core/telemetry/src/worker/node.rs

```diff
@@ -199,8 +225,23 @@ where TTrans::Output: Sink<BytesMut, Error = TSinkErr>
 				}
 
 				let item_len = item.len();
+				self.is_writing_data = true;
+				let mut timeout = Delay::new(Duration::from_millis(5000));
```

Comment:
I don't know if this PR is ready for review, but the `timeout` has to be in the struct, otherwise you're creating and destroying it immediately after.
`is_writing_data: bool` can probably be replaced with `timeout: Option<Delay>`.

Reply:
Thanks, it's ready for review now. I'm still wondering how to properly test this, as turning off the wifi doesn't seem to hit the timeout, for example.